src/app/teacher-session.service.ts

```diff
@@ -26,7 +26,7 @@ export class TeacherSessionService {
     switchMap(sessionId =>
       sessionId
         ? this.firebaseService.getStudentsInSession(sessionId)
-        : of(null)
+        : of(new Array<SessionStudentData>())
```

Comment:
This can be simplified, the type should be inferred automatically.
```suggestion
        : of([]])
```